src/node_crypto.h

```diff
@@ -95,6 +95,7 @@ class SecureContext : public BaseObject {
   SSL_CTX* ctx_;
   X509* cert_;
   X509* issuer_;
+  bool client_cert_engine_provided_ = false;
```

Comment:
Can you wrap this in an `#ifndef OPENSSL_NO_ENGINE` guard?  g++ will complain about it being unused otherwise (with `OPENSSL_NO_ENGINE` builds, that is.)